File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxans = Integer.MIN_VALUE;
        int j = 0;
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            maxans = Math.max(maxans, sum);
            if (sum < 0) {
                sum = 0;
                j++;
            }
        }
        return maxans;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        n=len(nums)
        maxans=float('-inf')
        maxele=0
        for x in nums:
            maxele+=x
            maxans=max(maxans,maxele)
            if maxele<0:
                maxele=0       
        return maxans              